File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit = 0
        lowest = prices[0]
        for i in range(1, len(prices)):
            maxProfit = max(maxProfit, prices[i] - lowest)
            if prices[i] < lowest:
                lowest = prices[i]
        return maxProfit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    maxProfit := 0
    minPrice := prices[0]
    for idx, price := range prices {
        if idx == 0 {
            continue
        }
        currProfit := price - minPrice
        if currProfit > maxProfit {
            maxProfit = currProfit
        }
        if price < minPrice {
            minPrice = price
        }
    }
    return maxProfit
}